dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeSwapDirectoriesTest.java

```diff
@@ -0,0 +1,162 @@
+	/*
+	 * Domains that are configured and cleared before and after the class.
+	 */
+	private static final String[] DOMAINS_ALL = { "domain1.com", "domain2.com", "domain3.com" };
+	private static final String[] DOMAINS_1 = { "domain1.com" };
+	private static final String[] DOMAINS_2 = { "domain1.com", "domain2.com", "domain3.com" };
+
+	public static CAContainer caContainer;
+	public static CAContainer caContainer2;
+
+	@Rule
+	public TestName testName = new TestName();
+
+	@BeforeClass
+	public static void beforeClass() throws Exception {
+		ORIGINAL_CONFIG = server.getServerConfiguration();
+		caContainer = new PebbleContainer();
+		caContainer2 = new PebbleContainer();
+		AcmeFatUtils.checkPortOpen(caContainer.getHttpPort(), 60000);
+		AcmeFatUtils.checkPortOpen(caContainer2.getHttpPort(), 60000);
+	}
+
+	@AfterClass
+	public static void afterClass() throws Exception {
+		if (caContainer != null) {
+			caContainer.stop();
+		}
+		if (caContainer2 != null) {
+			caContainer2.stop();
+		}
+	}
+
+	@After
+	public void afterTest() throws Exception {
+		/*
+		 * Clear the DNS records for the domain. Required since a few of the
+		 * tests setup invalid A records to test failure scenarios.
+		 */
+		AcmeFatUtils.clearDnsForDomains(caContainer, DOMAINS_ALL);
+		AcmeFatUtils.clearDnsForDomains(caContainer2, DOMAINS_ALL);
+
+		/*
+		 * Cleanup any generated ACME files.
+		 */
+		AcmeFatUtils.deleteAcmeFiles(server);
+	}
+
+	/**
+	 * Whether to use "acme://" style URIs, or "https://" style URIs.
+	 * 
+	 * <p/>
+	 * Sub-classes can override this method to determine which style URIs should
+	 * be used.
+	 * 
+	 * @return Whether to use "acme://" style URIs.
+	 */
+	protected boolean useAcmeURIs() {
+		return false;
+	}
+
+
+	@Test
+	@CheckForLeakedPasswords(AcmeFatUtils.CACERTS_TRUSTSTORE_PASSWORD)
+	public void update_directoryURI() throws Exception {
+
+		/*
+		 * Configure the acmeCA-2.0 feature.
+		 */
+		AcmeFatUtils.configureAcmeCA(server, caContainer, ORIGINAL_CONFIG, useAcmeURIs(), DOMAINS_1);
+
+		try {
+
+			/*
+			 * Start the server and wait for the certificate to be installed.
+			 */
+			server.startServer();
+			AcmeFatUtils.waitForAcmeToCreateCertificate(server);
+			AcmeFatUtils.waitForSslToCreateKeystore(server);
+			AcmeFatUtils.waitForSslEndpoint(server);
+
+			/*
+			 * Verify that the server is now using a certificate signed by the
+			 * CA.
+			 */
+			Certificate[] certificates1 = AcmeFatUtils.assertAndGetServerCertificate(server, caContainer);
+
+			/***********************************************************************
+			 * 
+			 * TEST: Update the directoryURI. This should result in a refreshed
+			 * certificate.
+			 * 
+			 **********************************************************************/
+			Log.info(this.getClass(), testName.getMethodName(), "TEST: START");
+			AcmeFatUtils.configureAcmeCA(server, caContainer2, ORIGINAL_CONFIG, useAcmeURIs(), DOMAINS_1);
+			AcmeFatUtils.waitForAcmeToCreateCertificate(server);
+
```

Comment:
Check the historical file.

Reply:
Not checking the historical file.

Reply:
Yes I am, lines 174-182.

Reply:
My bad, i misread the test and thought you were doing another operation immediately below.